### PR TITLE
adding Rocky 8

### PR DESCRIPTION
**Pull request by rgardner4012**
Adds Rocky 8 support:
* updates fixtures
*  updates functions/use_firewalld.pp, metadata.json, spec/classes/init.pp
* adds data/os/Rocky.yaml

### CODE REVIEW

**Comment by op-ct** [comment]
Code & test fixes look good.

With respect to the failing PR checks:

PR GLCI:
* GLCI check is expected to fail until simp/simp-core#889 is resolved, which affects the `main` branch of simp/github-action-gitlab-ci-syntax-check.
  * There is a fix in the https://github.com/simp/github-action-gitlab-ci-syntax-check/tree/tests/fix-gitlab-api branch (which also fixes the`set-output` failures).
  * That fix hasn't been rolled out in gitlab, because we prioritized the GHA replacement in simp/simp-core#887 
  * **TL;DR** ok to ignore for now as long as we've confirmed the acceptance tests work 
 
PR Tests:
* Puppet 8 / Ruby 3 spec matrix tests are also expected to fail until simp/simp-core#888 is resolved
  * If we're holding off on Puppet 8 support, we should either allow that test to fail or remove it from the test matrix
* Syntax check found a valid issue (hard tab)

**Review by michael-riddle — APPROVED** [review]
